react-native/react/util/carousel.desktop.js

```diff
@@ -0,0 +1,116 @@
+/* @flow */
+/* eslint-disable react/prop-types */
+
+import React from 'react'
+import _ from 'lodash'
+
+import {Component} from '../base-react'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import {Text} from '../common-adapters'
+
+export type CarouselProps = {
+  style?: ?Object,
+  itemWidth: number
+}
+
+class CarouselThing extends Component {
+  render (): ReactElement {
+    return (
+      <div style={{...this.props.style}}>
+          <Text style={{margin: 20}} type='Body'>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus sagittis lacus vel augue laoreet.</Text>
+      </div>
+    )
+  }
+}
+
+function PageIndicator ({selectedIndex, itemCount, style, onClickPage}: {selectedIndex: number, itemCount: number, style: Object, onClickPage: (i: number) => void}): ReactElement {
+  const marks = _.range(itemCount).map(n => {
+    const color = (n === selectedIndex) ? globalColors.white : globalColors.lightBlue
+    return (<i className='fa fa-circle' style={{color, ...styles.pageIndicator}} onClick={() => onClickPage(n)} key={n}/>)
+  })
+  return (
+    <div style={style}>{marks}</div>
+  )
+}
+
+export default class Carousel extends Component {
+  props: CarouselProps;
+  state: {selectedIndex: number, itemCount: number};
+
+  interval: number;
+  timeout: number;
+
+  constructor (props: CarouselProps) {
+    super(props)
+    this.state = {selectedIndex: 0, itemCount: 4}
+    this.timeout = 5e3
+  }
+
+  incrementIndex () {
+    this.setState({selectedIndex: (this.state.selectedIndex + 1) % this.state.itemCount})
+  }
+
+  componentDidMount () {
+    // timer to change the state
+    this.interval = setInterval(() => this.incrementIndex(), this.timeout)
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.interval)
+  }
+
+  resetIndex (i: number) {
+    this.setState({selectedIndex: i})
+    // Reset the timer so we don't change index as soon as the user clicks a page.
+    clearInterval(this.interval)
+    this.interval = setInterval(() => this.incrementIndex(), this.timeout)
+  }
+
+  render (): ReactElement {
+    const marginLeft = this.state.selectedIndex * -(this.props.itemWidth + 20)
+    const itemWidth = this.props.itemWidth
+    return (
+      <div style={{...styles.carousel, ...this.props.style}}>
+        <div style={{...globalStyles.flexBoxRow, width: itemWidth, overflow: 'visible'}}>
+          <div style={{...styles.carouselWrapper, marginLeft}}>
+            <CarouselThing style={{...styles.carouselThing, ...globalStyles.rounded, width: itemWidth}}/>
```

Comment:
Cool, no biggie either way. i just don't want someone else to pick it up and hard code it